app/views/users/new.html.haml

```diff
@@ -1,12 +1,39 @@
+<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
+
+<script type="text/javascript">
+Stripe.setPublishableKey("#{ENV['STRIPE_PUBLISHABLE_KEY']}");
+</script>
+
+= javascript_include_tag 'payment'
```

Comment:
Try to do this part as in solutions. `content_for :head do` ... You will see how to inject this part of the code to headers and make sure the code is there by checking in the browser. I think, you will be missing `payment.js` file as it was not included by asset pipeline. You can check also [my own project](https://github.com/tomtomecek/myflix/blob/master/app/views/users/new.html.haml#L1-L5).